### DISCUSSION TITLE
Does any one know Main class in the source.

### DISCUSSION THREAD

**Discussion by soshield** [DependencyTrack/dependency-track#2976] [General]
Code is build successfully, but when running dependency-track.war it is showing error "Missing Main attribute"
Also I checked MANIFEST.MF below is the content
Manifest-Version: 1.0
Build-Jdk-Spec: 17
Created-By: Maven Integration for Eclipse

**Comment by syalioune** [comment]
Hello @soshield 
Seems like you're trying to `java -jar ...` a `WAR` artifact which would not work. `WAR` can only be run by application servers like `Tomcat`.

You can follow https://github.com/DependencyTrack/dependency-track/blob/master/DEVELOPING.md to generate a proper `JAR` file that you can execute with `java -jar ...`

**Reply by soshield** [reply]
I ran "mvn clean package -P clean-exclude-wars -P enhance -P embedded-jetty -DskipTests -Dlogback.configuration.file=src/main/docker/logback.xml"

and getting this error

[INFO] --- jacoco:0.8.8:report (report) @ Dependency-Track ---
[INFO] Skipping JaCoCo execution due to missing execution data file.
[INFO] 
[INFO] --- exec:3.1.0:exec (merge-services-bom) @ Dependency-Track ---
[INFO] skipping execute as per configuration
[INFO] 
[INFO] --- antrun:3.1.0:run (deploy-bom) @ Dependency-Track ---
[INFO] Executing tasks
[INFO]      [copy] Copying 1 file to /Users/twilight/Documents/Dependency-Track/target/Dependency-Track/.well-known
[INFO] Executed tasks
[INFO] 
[INFO] --- jar:3.2.2:jar (default-jar) @ Dependency-Track ---
[INFO] Building jar: /Users/twilight/Documents/Dependency-Track/target/Dependency-Track.jar
[INFO] 
[INFO] --- antrun:3.1.0:run (rename-war-file) @ Dependency-Track ---
[INFO] Executing tasks
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  51.551 s
[INFO] Finished at: 2023-08-29T19:17:14+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.1.0:run (rename-war-file) on project Dependency-Track: An Ant BuildException has occured: Warning: Could not find file /Users/twilight/Documents/Dependency-Track/target/Dependency-Track-apiserver.war to copy.
[ERROR] around Ant part ...<move file="/Users/twilight/Documents/Dependency-Track/target/Dependency-Track-apiserver.war" tofile="/Users/twilight/Documents/Dependency-Track/target/Dependency-Track-apiserver.jar" />... @ 4:155 in /Users/twilight/Documents/Dependency-Track/target/antrun/build-main.xml
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException

**Reply by syalioune** [reply]
Hello,
Which version / branch are you trying to build ? So that I can try to reproduce your issue
